### PR TITLE
PR 1 of 3 TMB revamp: Separate TMB calculations as "coding only" and "all"

### PR DESCRIPTION
**Pull request by cansavvy**
#### What scientific question is your analysis addressing?

Currently the TMB stats calculated use all mutations, but we should also have a separate TMB stats for coding mutations only.

#### What was your approach?

Strelka and Mutect are only used for the "all mutations" TMB, but the consensus file of Strelka, Lancet, and Mutect is used for the "coding only" mutations this was discussed on #305 

This required some borrowing from the 02-merge-callers, and functionalization of `split_mnv` into its own script since it is now used by more than one script

I'm going to add the doc changes that are associated with this to PR #304 

#### What GitHub issue does your pull request address?

#305 

#### Which areas should receive a particularly close look?

-Is the original data handling flow of MNVs and SNVs maintained? I am particularly worried about the integrity of the last chunk of data wrangling on `03-calculate_tmb.R`.

- Are the `Variant Classification` groups of `IGR` and `Silent` the only ones that are considered as non-coding? 

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

No new tables yet, but this will require some downstream analyses like tmb_compare_tcga to be updated. Potentially mutational_signatures should be updated as well? 

#### What types of results are included (e.g., table, figure)?

No new results yet. 

#### Reproducibility Checklist

No new packages were needed here. 

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

### CODE REVIEW

File: analyses/snv-callers/scripts/03-calculate_tmb.R [hunk 1]
```diff
@@ -140,58 +149,115 @@ if (!all(unique(maf_df$Tumor_Sample_Barcode) %in% metadata$Tumor_Sample_Barcode)
...
-
+  
   # Set up BED region files for TMB calculations
   wgs_bed <- readr::read_tsv(opt$bed_wgs, col_names = FALSE)
   wxs_bed <- readr::read_tsv(opt$bed_wxs, col_names = FALSE)
-
+  
   # Calculate size of genome surveyed
   wgs_genome_size <- sum(wgs_bed[, 3] - wgs_bed[, 2])
   wxs_exome_size <- sum(wxs_bed[, 3] - wxs_bed[, 2])
```
[the remark this answers is not in the document]
Comment (cansavvy) [reply analyses/snv-callers/scripts/03-calculate_tmb.R]: Ah. Yea, I hadn't thought of this. It should change. I think what your describing above is what we should do for the `coding only` TMBs, but for the `all mutations` we should use the intersection of mutect and strelka's BED files and add that up. Does that sound right to you?

[the remark this answers is not in the document]
Comment (jashapiro) [reply analyses/snv-callers/scripts/03-calculate_tmb.R]: That does sound correct. I hadn't thought about that either... But yes, we can't call mutations outside the intersection of the two.

[the remark this answers is not in the document]
Comment (cansavvy) [reply analyses/snv-callers/scripts/03-calculate_tmb.R]: Follow up question is, what gtf file do we use?

[the remark this answers is not in the document]
Comment (cansavvy) [reply analyses/snv-callers/scripts/03-calculate_tmb.R]: Lancet methods list Gencode 31 as being used but I don't see the other callers list what Gencode version. I will dig deeper.

[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply analyses/snv-callers/scripts/03-calculate_tmb.R]: Gencode 27 is what is included in the data download at the moment.

[the remark this answers is not in the document]
Comment (jashapiro) [reply analyses/snv-callers/scripts/03-calculate_tmb.R]: The other callers didn't restrict to coding regions, so no GTFs for those...

[the remark this answers is not in the document]
Comment (cansavvy) [reply analyses/snv-callers/scripts/03-calculate_tmb.R]: Per our in person discussion, I will make the TMB denominator fixes in a subsequent PR so this PR doesn't get too out of control.

File: analyses/snv-callers/scripts/03-calculate_tmb.R [hunk 2]
```diff
@@ -140,58 +149,121 @@ if (!all(unique(maf_df$Tumor_Sample_Barcode) %in% metadata$Tumor_Sample_Barcode)
...
+                        dplyr::select(
+                          Tumor_Sample_Barcode,
+                          experimental_strategy,
+                          short_histology
+                        )) %>% 
+    dplyr::select(-index) 
+  
+  # Add in the MNVs
+  strelka_mutect_maf_df  %>%
+    dplyr::union_all(strelka_mutect_mnv) %>%
```
Comment (jashapiro) [inline analyses/snv-callers/scripts/03-calculate_tmb.R]: I think...
```suggestion
    dplyr::union_all(strelka_mutect_mnv, copy = TRUE) %>%
```

**Pull request merged**